src/debugging/netcore/NetCoreDebugHelper.ts

```diff
@@ -35,7 +35,16 @@ export interface NetCoreDebugScaffoldingOptions {
     appProject?: string;
 }
 
+export interface NetCoreProjectProperties {
+    assemblyName: string;
+    targetFramework: string;
+    appOutput: string;
+}
+
 export class NetCoreDebugHelper implements DebugHelper {
+
+    protected projectProperties: NetCoreProjectProperties | undefined;

```

Comment:
I'm not sure that having it as a class member will work in some cases. It is possible to have [multiple debug configurations launch at the same time](https://code.visualstudio.com/docs/editor/debugging#_compound-launch-configurations). If two launch configs were set up for debugging multiple containers (say, with compose up as a pre-launch task), whenever the first one got its project info, the second one would assume it's the same.

Additionally, if the project changes in some substantial way (say, the app output path changes), there's no mechanism to reset the cached project properties.

If possible, I would try to cache the values on the stack (i.e. as local variables) rather than as a class member, so that each time debugging happens, the particular debug call gets the information once.